src/utils.js

```diff
@@ -321,57 +321,48 @@ const getTaskData = (tasks, taskId) => {
  * @param {object[]} taskParameters
  * @param {object} operatorParameters
  * @param {object[]} featureOptions
+ * @param isDataset
  * @returns {object[]} configured operator parameters
  */
 const configureOperatorParameters = (
   taskParameters,
   operatorParameters,
-  featureOptions
+  featureOptions,
+  isDataset
 ) => {
-  const configuredOperatorParameters = taskParameters.map((parameter) => {
-    let options;
-    if (parameter.options) {
-      // use parameter options if exist
-      options = parameter.options;
-    } else if (parameter.type === 'feature') {
-      // use feature options if parameter is type feature
-      options = featureOptions;
-    } else {
-      options = undefined;
-    }
-
-    let value;
-    if (parameter.name in operatorParameters) {
-      /**
-       *  get the value from operatorParameters if parameter exist in operatorParameters
-       *  to parameter multiple we need to split string by ','
-       *  because the multiple values are in this format 'value1,value2,value3'
-       */
-      if (parameter.multiple) {
-        value = operatorParameters[parameter.name].split(',').filter((el) => {
-          return el !== '';
-        });
-      } else {
-        value = operatorParameters[parameter.name];
-      }
-    } else if (parameter.multiple) {
-      // for parameter multiple use a empty array to default value
-      value = [];
-    } else if (parameter.type === 'feature') {
-      // for feature parameter use null to default value
-      value = null;
-    } else {
-      value = parameter.default;
-    }
-
-    return {
-      ...parameter,
-      options: options,
-      value: value,
-    };
-  });
+  const datasetParameters =
+    isDataset && operatorParameters
+      ? Object.keys(operatorParameters).map((key) => ({
+          name: key,
+          value: operatorParameters[key],
+        }))
+      : undefined;
+
+  let configuredOperatorParameters;
+
+  if (!isDataset)
+    configuredOperatorParameters = taskParameters.map((parameter) => {
+      return {
+        ...parameter,
+        options: parameter.options
+          ? parameter.options
+          : parameter.type === 'feature'
+          ? featureOptions
+          : undefined,
+        value:
+          parameter.name in operatorParameters
+            ? parameter.type === 'feature' || parameter.multiple
+              ? operatorParameters[parameter.name].split(',').filter((el) => {
+                  return el !== '';
+                })
+              : operatorParameters[parameter.name]
+            : parameter.type === 'feature' || parameter.multiple
+            ? []
+            : parameter.default,
+      };
+    });
 
-  return configuredOperatorParameters;
+  return datasetParameters || configuredOperatorParameters;
```

Comment:
@AlendeGeraldo este trecho de código tem muitos operadores ternários `?` e `:` e nenhum comentário.
Não dá pra entender nada...

Por favor, substitua os `?` `:` por `if` e `else`. Use-os somente se for 1 condição. Ternário dentro de ternário nunca.

Veja que havia vários comentários que eu tinha pedido pro Shiguemi colocar aí para deixar este código mais explicativo e eles foram apagados. Por que?